extension/extension_tests/url-test.js

```diff
@@ -14,54 +14,75 @@ function addResult(testInput, testOutput, passed) {
 }
 
 /**
- * Check whether the ReplaceUrl method gives the expected output.
+ * Check whether the GetMinimalUrl method gives the expected output.
  * @param {string} testInput The input URL.
  * @param {string} expectedOutput The anticipated output URL.
- * @returns {boolean} Whether they result matches the output.
+ * @returns {boolean} Whether the result matches the output.
  */
-function runTest(testInput, expectedOutput) {
+function runMinimizerTest(testInput, expectedOutput) {
   var actualOutput = urlHandler.GetMinimalUrl(testInput);
   addResult(testInput, actualOutput, actualOutput === expectedOutput);
 }
 
 /**
+ * Check whether the ReplaceUrl method gives the expected output.
+ * @param {string} testInput The input URL.
+ * @param {bool} expectedOutput The anticipated result.
+ * @returns {boolean} Whether the result matches the output.
+ */
+function runGreenLockTest(testInput, expectedOutput) {
+  var actualOutput = urlHandler.IsGreenLockSite(testInput);
+  addResult(testInput, actualOutput, actualOutput === expectedOutput);
+}
+
+/**
  * Runs a set of tests with URLs.
  */
 function runTests() {
   // Basic origin.
-  runTest('www.example.com', 'www.example.com');
+  runMinimizerTest('www.example.com', 'www.example.com');
 
   // Origin with a trailing slash.
-  runTest('www.example.com/', 'www.example.com');
+  runMinimizerTest('www.example.com/', 'www.example.com');
 
   // Origin with a simple path.
-  runTest('www.example.com/path', 'www.example.com');
+  runMinimizerTest('www.example.com/path', 'www.example.com');
 
   // Origin with a longer path.
-  runTest('www.example.com/path.path/path/', 'www.example.com');
+  runMinimizerTest('www.example.com/path.path/path/', 'www.example.com');
 
   // Origin with a mix of numbers and letters.
-  runTest('www.exa123mple.com', 'www.exa123mple.com');
+  runMinimizerTest('www.exa123mple.com', 'www.exa123mple.com');
 
   // Origin with scheme.
-  runTest('https://www.example.com', 'www.example.com');
+  runMinimizerTest('https://www.example.com', 'www.example.com');
 
   // Origin with scheme.
-  runTest('http://www.example.com', 'www.example.com');
+  runMinimizerTest('http://www.example.com', 'www.example.com');
 
   // Origin with non-standard scheme.
-  runTest('file://www.example.com', 'file://www.example.com');
-
-  // Origin with missing scheme.
-  runTest('://www.example.com', '');
+  runMinimizerTest('file://www.example.com', 'file://www.example.com');
 
   // Origin with standard port.
-  runTest('http://www.example.com:80', 'www.example.com');
+  runMinimizerTest('http://www.example.com:80', 'www.example.com');
 
   // Origin with non-standard port.
-  runTest('http://www.example.com:3000', 'www.example.com:3000');
+  runMinimizerTest('http://www.example.com:3000', 'www.example.com:3000');
+
+  // Check that whitelisting works.
+  runGreenLockTest('https://www.google.com', true);
+  runGreenLockTest('https://google.com', true);
```

Comment:
I did that intentionally bc I want to catch all google.co.uk, google.co.in, etc. without adding a super long (and possibly incorrect) list. I figure google.example.com will be rare. What do you think of that decision? I added a test & comment to document.